### PR TITLE
refactor: replace unnecessary use of `default_factory` with `default`

### PR DESCRIPTION
**Pull request by wer153**
PR replaces unnecessary use of `default_factory` with `default` for dataclass fields on the `LoggingMiddlewareConfig` where the default value returned by the factory requires no computation and is immutable.

### Description
As far as I understand, By specifying a factory function for a mutable default value, each instance gets its own independent copy of the default value, `preventing unintended sharing of mutable state between instances`. This helps in maintaining encapsulation and avoiding potential side effects that could arise from shared mutable state.

However, in the code, `default_factory was not used for such a purpose`. Therefore, I suggest `using default to avoid misinterpretation of usage`.

### CODE REVIEW

File: litestar/middleware/logging.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ class LoggingMiddlewareConfig:
             Thus, re-arranging the log-message is as simple as changing the iterable.
         -  To turn off logging of responses, use and empty iterable.
     """
-    middleware_class: type[LoggingMiddleware] = field(default_factory=lambda: LoggingMiddleware)
+    middleware_class: type[LoggingMiddleware] = LoggingMiddleware
```
Comment (cbscsm) [inline litestar/middleware/logging.py:308]: How about use `field` function for consistency.
```suggestion
    middleware_class: type[LoggingMiddleware] = field(default=LoggingMiddleware)
```

Comment (wer153) [reply litestar/middleware/logging.py:308]: Makes sense. Thanks for the suggestion!

**Review by peterschutt — APPROVED** [review]
LGTM, thanks!

**Comment by wer153** [comment]
https://github.com/all-contributors add @wer153 for code

**Comment by wer153** [comment]
@all-contributors add @wer153 for code

**Pull request merged**